src/dbnode/server/server.go

```diff
@@ -564,9 +574,20 @@ func Run(runOpts RunOptions) {
 
 	if cfg.DebugListenAddress != "" {
 		go func() {
-			if err := http.ListenAndServe(cfg.DebugListenAddress, nil); err != nil {
+			mux := http.DefaultServeMux
+			if debugWriter != nil {
+				if err = debugWriter.RegisterHandler("/debug/dump", mux); err != nil {
```

Comment:
you can still use `:=` (to avoid overwriting the global) since its block scoped

Reply:
Fixed